components/camel-nats/src/main/java/org/apache/camel/component/nats/NatsProducer.java

```diff
@@ -17,19 +17,29 @@
 package org.apache.camel.component.nats;
 
 import java.time.Duration;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
 
 import io.nats.client.Connection;
 import io.nats.client.Connection.Status;
+import io.nats.client.Message;
+import org.apache.camel.AsyncCallback;
 import org.apache.camel.Exchange;
-import org.apache.camel.support.DefaultProducer;
+import org.apache.camel.ExchangeTimedOutException;
+import org.apache.camel.InvalidPayloadException;
+import org.apache.camel.support.DefaultAsyncProducer;
 import org.apache.camel.util.ObjectHelper;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-public class NatsProducer extends DefaultProducer {
+public class NatsProducer extends DefaultAsyncProducer {
 
     private static final Logger LOG = LoggerFactory.getLogger(NatsProducer.class);
 
+    private final ScheduledExecutorService scheduler = Executors.newScheduledThreadPool(1);
```

Comment:
We should use Camle's ExecutorStrategy to get a thread pool so its properly started/stopped etc. Also it may be desiredable to allow to configure a higher pool size, in case there is a lot of timed out messages. Otherwise the tasks is piling up as its the timeout future thread that will be used for continued routing.

Reply:
I'm using camel's `ExecutorServiceManager` now, does that look correct?